src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -705,22 +710,23 @@ private static ConcurrentQueue<BlockCommitSet> CreateQueueAtomic(ref ConcurrentQ
             return prior ?? instance;
         }
 
-        private void CreateCommitSet(long blockNumber)
+        private BlockCommitSet CreateCommitSet(long blockNumber)
         {
             if (_logger.IsDebug) _logger.Debug($"Beginning new {nameof(BlockCommitSet)} - {blockNumber}");
 
             // TODO: this throws on reorgs, does it not? let us recreate it in test
-            Debug.Assert(CurrentPackage is null || blockNumber == CurrentPackage.BlockNumber + 1, "Newly begun block is not a successor of the last one");
-            Debug.Assert(IsCurrentListSealed, "Not sealed when beginning new block");
+            Debug.Assert(!(_commitSetQueue ?? CreateQueueAtomic(ref _commitSetQueue)).TryDequeue(out BlockCommitSet lastSet) || blockNumber == lastSet.BlockNumber + 1, $"Newly begun block is not a successor of the last one.");
```

Comment:
why _commitSetQueue can't be:
a. Created at start
b. Created lazily through property

Would simplify the code a bit.

Reply:
Won't the Dequeue break the queue?

Reply:
Ow shit, this is suppose to be peek.